### PR TITLE
C2LC-227: Added check for unused function parameters.

### PR DESCRIPTION
**Pull request by the-t-in-rtf**
See [C2LC-227](https://issues.fluidproject.org/browse/C2LC-227) for details.

cc: @simonbates

### CODE REVIEW

**Comment by the-t-in-rtf** [comment]
This is intentionally broken at the moment so we can discuss what/how to fix things.  Here's are the linting failures with the new rule:

```
/Users/duhrer/Source/rtf/c2lc-coding-environment/src/App.js
  102:46  error  'reject' is defined but never used     no-unused-vars
  120:46  error  'reject' is defined but never used     no-unused-vars
  138:46  error  'reject' is defined but never used     no-unused-vars
  405:24  error  'prevProps' is defined but never used  no-unused-vars

/Users/duhrer/Source/rtf/c2lc-coding-environment/src/DashDriver.js
  34:38  error  'reject' is defined but never used  no-unused-vars

/Users/duhrer/Source/rtf/c2lc-coding-environment/src/FakeRobotDriver.js
   6:13  error  'onDisconnected' is defined but never used  no-unused-vars
  11:38  error  'reject' is defined but never used          no-unused-vars

/Users/duhrer/Source/rtf/c2lc-coding-environment/src/FocusTrapManager.test.js
  8:5  error  'trapCloseHandler' is defined but never used  no-unused-vars

/Users/duhrer/Source/rtf/c2lc-coding-environment/src/Interpreter.test.js
    8:38  error  'reject' is defined but never used       no-unused-vars
  162:53  error  'interpreter' is defined but never used  no-unused-vars
  163:38  error  'reject' is defined but never used       no-unused-vars
  180:53  error  'interpreter' is defined but never used  no-unused-vars
  181:38  error  'reject' is defined but never used       no-unused-vars
  187:53  error  'interpreter' is defined but never used  no-unused-vars
  188:38  error  'reject' is defined but never used       no-unused-vars
  207:38  error  'reject' is defined but never used       no-unused-vars
  215:53  error  'interpreter' is defined but never used  no-unused-vars

/Users/duhrer/Source/rtf/c2lc-coding-environment/src/ProgramBlockEditor.test.js
  138:64  error  'index' is defined but never used  no-unused-vars
```

The very last one is the problem fixed in my aria labels pull.  The others are things we should discuss, I'd propose looking through it together in our next tech chat.  We may decide to add hints to ignore things, to add tests that exercise the parameters, or just to remove them.  I still think this is a fairly small pull once we agree on what to do, but need 15 minutes of feedback before I take it on.

**Comment by the-t-in-rtf** [comment]
Sorry, accidentally committed the test cleanup work here and had to revert that.

File: src/Interpreter.test.js [hunk 1]
```diff
@@ -22,10 +22,10 @@ function createInterpreter() {
 function createMockCommandHandler() {
     const mockCommandHandler = jest.fn();
     mockCommandHandler.mockImplementation(() => {
-        return new Promise((resolve, reject) => {
+        return new Promise((resolve) => {
             setTimeout(() => {
                 resolve();
-            }, 0);
+            }, 250);
```
Comment (sbates-idrc) [inline src/Interpreter.test.js:28]: Do we need this change to the timeout value?

**Comment by sbates-idrc** [comment]
@the-t-in-rtf thanks for this PR. I've left one small comment on the change in the timeout value for the command handler returned from `createMockCommandHandler()`.

Also, this PR changes `package-lock.json`, which I believe we don't need as our dependencies haven't changed on this branch. I made C2LC-227 branch with the `package-lock.json` from `develop-0.8`: https://github.com/sbates-idrc/c2lc-coding-environment/commit/bf073b7970e25ee71d6a5cc4855eaba941a3d1ab

File: src/Interpreter.test.js (hunk 1 above)
Comment (the-t-in-rtf) [reply src/Interpreter.test.js]: IIRC that was left over from an early attempt to fix the race issues.  I reverted it.

**Comment by the-t-in-rtf** [comment]
> Also, this PR changes `package-lock.json`, which I believe we don't need as our dependencies haven't changed on this branch. I made a C2LC-227 branch with the `package-lock.json` from `develop-0.8`: [sbates-idrc@bf073b7](https://github.com/sbates-idrc/c2lc-coding-environment/commit/bf073b7970e25ee71d6a5cc4855eaba941a3d1ab)

Done.

**Pull request merged**